xbmc/Application.cpp

```diff
@@ -3210,7 +3210,8 @@ PlayBackRet CApplication::PlayStack(const CFileItem& item, bool bRestart)
       }
     }
 
-    *m_itemCurrentFile = item;
+    CFileItemPtr tmp(new CFileItem(item));
```

Comment:
couldn't you use CApplication::SetCurrentFileItem(item) here (and below) instead?